### PR TITLE
[Skylark] Avoid unnecessary subList invocations.

### PR DESCRIPTION
**Pull request by ttsugriy**
It's better to not add `self` positional argument if it's not needed, instead
of adding it and removing it laster on. `subList` invoaction results in more
allocations and CPU overhead when using positional args.

### CODE REVIEW

**Comment by laurentlb** [comment]
LGTM, thanks!

File: src/main/java/com/google/devtools/build/lib/syntax/FuncallExpression.java [hunk 1]
```diff
@@ -731,26 +731,22 @@ private boolean includeSelfAsArg(
   /**
    * Call a method depending on the type of an object it is called on.
    *
-   * @param positionals The first object is expected to be the object the method is called on.
+   * @param positionalArgs The first object is expected to be the object the method is called on.
```
Comment (c-parsons) [inline src/main/java/com/google/devtools/build/lib/syntax/FuncallExpression.java]: This is no longer necessarily true. Could you please update the doc?

Comment (c-parsons) [inline src/main/java/com/google/devtools/build/lib/syntax/FuncallExpression.java]: This is no longer necessarily true. Could you please update the doc?

Comment (ttsugriy) [reply src/main/java/com/google/devtools/build/lib/syntax/FuncallExpression.java]: good catch!

**Comment by ttsugriy** [comment]
updated javadoc

**Pull request closed without merging**